src/reflect/scala/reflect/internal/Types.scala

```diff
@@ -4030,13 +4030,14 @@ trait Types
         if (isIntersectionTypeForLazyBaseType(original)) intersectionTypeForLazyBaseType(parents)
         else refinedType(parents, owner)
       val syms1 = decls.toList
-      for (sym <- syms1)
-        result.decls.enter(sym.cloneSymbol(result.typeSymbol).resetFlag(OVERRIDE))
-      val syms2 = result.decls.toList
```

Comment:
It seems preferable to get the `List[Symbol]` from `Scope#toList`, as that would be internally cached in `Scope` and available for later calls to `toList`.

I would also guard all this code with `if (!decls.isEmpty)` to avoid (as before) allocation of the `TypeMaps` for refinement types that don't have decls (like `A with B` or `T {} `)

Otherwise this change LGTM. We know by construction that the fast path of `syms1 == syms2` in `substThisAndSym` can never apply

Reply:
Changes applied. Could you take another look?